outputs.tf

```diff
@@ -1,93 +1,106 @@
-  this_autoscaling_group_default_cooldown          = concat(aws_autoscaling_group.this.*.default_cooldown, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.default_cooldown, [""])[0]
-  this_autoscaling_group_health_check_grace_period = concat(aws_autoscaling_group.this.*.health_check_grace_period, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.health_check_grace_period, [""])[0]
-  this_autoscaling_group_health_check_type         = concat(aws_autoscaling_group.this.*.health_check_type, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.health_check_type, [""])[0]
-  this_autoscaling_group_availability_zones        = concat(aws_autoscaling_group.this.*.availability_zones, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.availability_zones, [""])[0]
-  this_autoscaling_group_vpc_zone_identifier       = concat(aws_autoscaling_group.this.*.vpc_zone_identifier, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.vpc_zone_identifier, [""])[0]
-  this_autoscaling_group_load_balancers            = concat(aws_autoscaling_group.this.*.load_balancers, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.load_balancers, [""])[0]
-  this_autoscaling_group_target_group_arns         = concat(aws_autoscaling_group.this.*.target_group_arns, aws_autoscaling_group.this_with_initial_lifecycle_hook.*.target_group_arns, [""])[0]
-}
-
-output "this_launch_configuration_id" {
+################################################################################
+# Launch configuration
+################################################################################
+
+output "launch_configuration_id" {
   description = "The ID of the launch configuration"
-  value       = local.this_launch_configuration_id
+  value       = element(concat(aws_launch_configuration.this.*.id, [""]), 0)
+}
+
+output "launch_configuration_arn" {
+  description = "The ARN of the launch configuration"
+  value       = element(concat(aws_launch_configuration.this.*.arn, [""]), 0)
 }
 
-output "this_launch_configuration_name" {
+output "launch_configuration_name" {
   description = "The name of the launch configuration"
-  value       = local.this_launch_configuration_name
+  value       = element(concat(aws_launch_configuration.this.*.name, [""]), 0)
+}
+
+################################################################################
+# Launch template
+################################################################################
+
+output "launch_template_id" {
+  description = "The ID of the launch template"
+  value       = element(concat(aws_launch_template.this.*.id, [""]), 0)
 }
 
-output "this_autoscaling_group_id" {
+output "launch_template_arn" {
+  description = "The ARN of the launch template"
+  value       = element(concat(aws_launch_template.this.*.arn, [""]), 0)
+}
+
+output "launch_template_latest_version" {
```

Comment:
This should be `launch_template_name`.

I have mixed feelings about dropping "this_" in names. Legacy from pre-0.12-times, I know.

One part of me still wants this_ prefix but another part of me can't find enough reasons for having that in this specific module :)

Do you drop it on purpose, or just not? :)

Reply:
good catch, fixed in https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/139/commits/62b108510de3cc70095fe24f855b863af399ebbf

I actually dropped it intentionally here - I was hoping to get away from it because its just additional typing without any real value